### PR TITLE
fix: armies updates on worldmap

### PR DESCRIPTION
**Pull request by edisontim**

### CODE REVIEW

File: client/src/three/components/ArmyManager.ts [hunk 1]
```diff
@@ -97,65 +89,94 @@ export class ArmyManager {
   async onUpdate(update: ArmySystemUpdate) {
     await this.loadPromise;
 
-    const { entityId, hexCoords, isMine } = update;
+    // console.log(
```
Comment (cwastche) [inline client/src/three/components/ArmyManager.ts:92]: Remove comments

File: client/src/three/components/ArmyManager.ts [hunk 2]
```diff
@@ -97,65 +89,94 @@ export class ArmyManager {
...
+    //   `[MyApp] onUpdate ${JSON.stringify(
+    //     update,
+    //     (key, value) => (typeof value === "bigint" ? value.toString() : value),
+    //     2,
+    //   )}`,
+    // );
+    const { entityId, hexCoords, isMine, health, battleId, defender } = update;
+
+    if (defender) {
+      //   console.log(`[MyApp] army is defender ${entityId}`);
```
Comment (cwastche) [inline client/src/three/components/ArmyManager.ts:102]: Remove comment

File: client/src/three/components/ArmyManager.ts [hunk 3]
```diff
@@ -97,65 +89,94 @@ export class ArmyManager {
...
+      return;
+    }
+
+    if (battleId !== 0) {
+      if (this.armies.has(entityId)) {
+        // console.log(`[MyApp] removing army ${entityId}`);
+        this.removeArmy(entityId);
+        return;
+      } else {
+        // console.log(`[MyApp] skipping for army ${entityId}`);
```
Comment (cwastche) [inline client/src/three/components/ArmyManager.ts:112]: Remove comments

File: client/src/three/components/ArmyManager.ts [hunk 4]
```diff
@@ -97,65 +89,94 @@ export class ArmyManager {
...
 
-    const normalizedCoord = { col: hexCoords.col - FELT_CENTER, row: hexCoords.row - FELT_CENTER };
+    const position = new Position({ x: hexCoords.col, y: hexCoords.row });
 
     if (this.armies.has(entityId)) {
-      this.moveArmy(entityId, normalizedCoord);
+      this.moveArmy(entityId, position);
     } else {
-      this.addArmy(entityId, normalizedCoord, isMine);
+      //   console.log(`[MyApp] adding army ${entityId}`);
```
Comment (cwastche) [inline client/src/three/components/ArmyManager.ts:122]: Remove comment

File: client/src/three/systems/SystemManager.ts [hunk 5]
```diff
@@ -29,20 +29,32 @@ export class SystemManager {
...
+          const protectee = getComponentValue(this.dojo.components.Protectee, update.entity);
+
+          const healthMultiplier =
+            EternumGlobalConfig.troop.healthPrecision * BigInt(EternumGlobalConfig.resources.resourcePrecision);
 
           return {
             entityId: army.entity_id,
             hexCoords: this.getHexCoords(update.value),
             isMine,
+            health: {
```
Comment (aymericdelab) [inline client/src/three/systems/SystemManager.ts:52]: why do you need health ? not used in ArmyManager

File: client/src/ui/modules/entity-details/EntityDetails.tsx [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import { useMemo } from "react";
 import { useLocation } from "wouter";
-import { CombatEntityDetails } from "./CombatEntityDetails";
 import { BuildingEntityDetails } from "./BuildingEntityDetails";
+import { CombatEntityDetails } from "./CombatEntityDetails";
```
Comment (aymericdelab) [inline client/src/ui/modules/entity-details/EntityDetails.tsx:4]: are we using different linter ? why are there some places where lines order are changed. ?

**Pull request merged**